### PR TITLE
Make accelerate install non-torch dependent

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Removes `accelerate` deps from eager install, which will upgrade pytorch dep. Should be okay as I think all of the required deps are installed normally with transformers anyways. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts @ydshieh

### CODE REVIEW

File: .circleci/create_circleci_config.py [hunk 1]
```diff
@@ -285,7 +285,8 @@ def job_name(self):
         "pip install --upgrade --upgrade-strategy eager pip",
         "pip install -U --upgrade-strategy eager .[sklearn,tf-cpu,torch,testing,sentencepiece,torch-speech,vision]",
         "pip install -U --upgrade-strategy eager tensorflow_probability",
-        "pip install -U --upgrade-strategy eager -e git+https://github.com/huggingface/accelerate@main#egg=accelerate",
+        # Without this we can't pin torch versions in the future
```
Comment (amyeroberts) [inline .circleci/create_circleci_config.py:288]: I'd make this more general - there might be libraries other than torch we both touch 

```suggestion
        # Without this we can't dependency versions in the future
```

Comment (ydshieh) [inline .circleci/create_circleci_config.py:288]: maybe replace `this` with `--no-deps` and explain what `we can't pin torch versions in the future` means or replace it with a more detailed sentence?

**Review by amyeroberts — APPROVED** [review]
Thanks! 

I expect this to cause some pain in the future if accelerate changes its package requirements requiring a min version. Might be useful for flagging incompatibilities between the two instead of silently changing the installs

File: .circleci/create_circleci_config.py [hunk 2]
```diff
@@ -285,7 +285,8 @@ def job_name(self):
         "pip install --upgrade --upgrade-strategy eager pip",
         "pip install -U --upgrade-strategy eager .[sklearn,tf-cpu,torch,testing,sentencepiece,torch-speech,vision]",
         "pip install -U --upgrade-strategy eager tensorflow_probability",
-        "pip install -U --upgrade-strategy eager -e git+https://github.com/huggingface/accelerate@main#egg=accelerate",
+        # Without this we can't dependency versions in the future
```
Comment (amyeroberts) [inline .circleci/create_circleci_config.py:288]: d'oh sorry - bad original suggestion 

```suggestion
        # Without this we can't pin dependency versions in the future
```

File: .circleci/create_circleci_config.py (hunk 1 above)
Comment (ydshieh) [reply .circleci/create_circleci_config.py]: missing a `pin`

File: .circleci/create_circleci_config.py (hunk 2 above)
Comment (muellerzr) [reply .circleci/create_circleci_config.py]: no worries, I forgot to get it in my expound too :)

Comment (ydshieh) [reply .circleci/create_circleci_config.py]: I guess you somehow forgot to change to `dependency versions` (unless that is what you intended to)

**Comment by ydshieh** [comment]
I guess you somehow forgot to change to dependency versions (unless that is what you intended to)

**Comment by muellerzr** [comment]
@ydshieh in the mix this just needs to be rebased, should be fine :)

**Comment by ydshieh** [comment]
I mean `Without --no-deps we can't pin PyTorch versions in the future` --> `Without --no-deps we can't pin dependency versions in the future` (@amyeroberts 's comment).

**Review by ydshieh — APPROVED** [review]
Thank you!

**Pull request merged**